### ISSUE TITLE
`tests/integration_tests/test_integration.py` is failing on macOS

### ISSUE THREAD

**Issue by hvy** [optuna/optuna#2695]
## Motivation

Although macOS is not officially supported, the CI tests should ideally not be failing. Currently, integration tests for macOS are.

### Example Failure from Daily CI

```
tests/integration_tests/test_integration.py ..
[I 2021-05-21 09:51:46,390] A new study created in memory with name: no-name-85517cbd-9f43-4fb8-be2d-0c1d828a8ab2
2021-05-21 09:51:46.396531: I tensorflow/compiler/jit/xla_cpu_device.cc:41] Not creating XLA devices, tf_xla_enable_xla_devices not set
2021-05-21 09:51:46.459935: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:116] None of the MLIR optimization passes are enabled (registered 2)
[I 2021-05-21 09:51:46,854] Trial 0 pruned. Trial was pruned at epoch 0.
[I 2021-05-21 09:51:46,854] A new study created in memory with name: no-name-bb2bc76b-0901-4798-afda-8118aea997d0
[I 2021-05-21 09:51:47,268] Trial 0 finished with value: 1.0 and parameters: {}. Best is trial 0 with value: 1.0.
[I 2021-05-21 09:51:47,271] A new study created in memory with name: no-name-b04903bc-b115-458e-9954-805fd5acaada
[I 2021-05-21 09:51:47,704] Trial 0 finished with value: 1.0 and parameters: {}. Best is trial 0 with value: 1.0.
[I 2021-05-21 09:51:47,704] A new study created in memory with name: no-name-dce11dea-fecf-4dbb-a086-6bbd5026a6d5
[I 2021-05-21 09:51:48,127] Trial 0 finished with value: 1.0 and parameters: {}. Best is trial 0 with value: 1.0.
[I 2021-05-21 09:51:48,129] A new study created in memory with name: no-name-fdf093db-7ca7-4722-82a1-4dbb67698522
[I 2021-05-21 09:51:48,550] Trial 0 pruned. Trial was pruned at epoch 1.
[I 2021-05-21 09:51:48,551] A new study created in memory with name: no-name-8720da64-bcf1-476a-b0f2-9cc04b6fe117
[I 2021-05-21 09:51:48,955] Trial 0 finished with value: 1.0 and parameters: {}. Best is trial 0 with value: 1.0.
[I 2021-05-21 09:51:48,956] A new study created in memory with name: no-name-724a6da9-0fa3-4187-af45-e9b7ae29f3b3
[I 2021-05-21 09:51:48,958] A new study created in memory with name: no-name-2d9c26d7-ff2c-4afd-b761-ab49fe46cb92
tests/integration_tests/test_keras.py .....
[I 2021-05-21 09:51:48,960] A new study created in memory with name: no-name-bb1a4f5d-f609-48d1-aead-36ad6e88230a
[I 2021-05-21 09:51:48,961] A new study created in memory with name: no-name-ba303d0a-9ea7-4dd4-8beb-11222f7e125f
[I 2021-05-21 09:51:48,962] A new study created in memory with name: no-name-64950f3a-0ac5-4141-b496-3a6ec378c929
[I 2021-05-21 09:51:48,963] A new study created in memory with name: no-name-d20bba17-4ee8-4b08-9cb5-8518d077190a
[I 2021-05-21 09:51:48,965] A new study created in memory with name: no-name-e46220e3-85d8-4016-9b5d-7f4e911e7ec3
Fatal Python error: Segmentation fault

Thread 0x0000700009525000 (most recent call first):

Thread 0x000000010c03adc0 (most recent call first):
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/lightgbm/basic.py", line 1392 in __init_from_csr
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/lightgbm/basic.py", line 1278 in _lazy_init
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/lightgbm/basic.py", line 1472 in construct
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/lightgbm/basic.py", line 2229 in __init__
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/lightgbm/engine.py", line 228 in train
  File "/Users/runner/work/optuna/optuna/tests/integration_tests/test_lightgbm.py", line 145 in objective
tests/integration_tests/test_lightgbm.py ..
  File "/Users/runner/work/optuna/optuna/optuna/_optimize.py", line 216 in _run_trial
  File "/Users/runner/work/optuna/optuna/optuna/_optimize.py", line 162 in _optimize_sequential
  File "/Users/runner/work/optuna/optuna/optuna/_optimize.py", line 75 in _optimize
  File "/Users/runner/work/optuna/optuna/optuna/study.py", line 410 in optimize
  File "/Users/runner/work/optuna/optuna/tests/integration_tests/test_lightgbm.py", line 51 in test_lightgbm_pruning_callback
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/python.py", line 183 in pytest_pyfunc_call
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/python.py", line 1641 in runtest
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/runner.py", line 162 in pytest_runtest_call
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/runner.py", line 255 in <lambda>
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/runner.py", line 311 in from_call
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/runner.py", line 255 in call_runtest_hook
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/runner.py", line 215 in call_and_report
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/runner.py", line 126 in runtestprotocol
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/runner.py", line 109 in pytest_runtest_protocol
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/main.py", line 348 in pytest_runtestloop
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/main.py", line 323 in _main
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/main.py", line 269 in wrap_session
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/main.py", line 316 in pytest_cmdline_main
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/config/__init__.py", line 163 in main
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/_pytest/config/__init__.py", line 185 in console_main
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/bin/pytest", line 8 in <module>
/Users/runner/work/_temp/b8af8ec8-ed83-413a-a30a-02d1720921e9.sh: line 2:  3388 Segmentation fault: 11  pytest -s tests/integration_tests
Error: Process completed with exit code 139.

```

https://github.com/optuna/optuna/runs/2638103566

**Comment by nzw0301** [comment]
Let me share a related issue: https://github.com/pytest-dev/pytest/issues/3672.

**Comment by nzw0301** [comment]
I think the issue is resolved by #2896. Let me close this issue.
